### ISSUE TITLE
Analyzer no longer auto-detects the package root

### ISSUE THREAD

**Issue by nex3** [dart-lang/sdk#23928]
As of a recent commit in bleeding edge, the analyzer doesn't seem to automatically detect the package root anymore. `package:` imports all report "Target of URI does not exist" unless `--packages` or `--package-root` is passed in explicitly.

**Comment by pq** [comment]
This is in the `analyzer_cli`.  I think it surfaced as of this DEPS update: https://github.com/dart-lang/sdk/commit/89b4359f2373def9385a6982aa26488281c57881.  (It's a plausible guess but I haven't bisected.)

**Comment by pq** [comment]
Trying to repro with `test`.  Specifically:

```
[~/src/git/clones/test] (master) $ dartanalyzer --version
dartanalyzer version 1.12.0-edge.868109bf97d05a8e20b5b5a8754755b9f4e72cc1
[~/src/git/clones/test]  (master) $ dartanalyzer lib/src/executable.dart
Analyzing [lib/src/executable.dart]...
[hint] 'description' is deprecated (/Users/me/src/git/clones/test/lib/src/frontend/future_matchers.dart, line 34, col 10)
1 hint found.
```

**Comment by pq** [comment]
FWIW: I'm attempting to repro on Mac.  @nex3 is on Linux.  I'm also using a freshly built SDK.

**Comment by nex3** [comment]
I can no longer reproduce this using a fresh SDK, so I guess it was accidentally fixed some time between cc1b7f0c3e0fbd39ed0f24d2af7e2d25d1aa9dd0 and 64ff591f56a6248afdd29470cab64b86133c75eb.
